rediscluster/nodemanager.py

```diff
@@ -279,14 +279,18 @@ def remap_internal_node_object(self, node_obj):
                     # print('remapping host', node_obj[0], remap_rule['to_host'])
                     node_obj[0] = remap_rule['to_host']
 
-            ## The port value is always an integer
-            if 'from_port' in remap_rule and 'to_port' in remap_rule:
-                if remap_rule['from_port'] == node_obj[1]:
-                    # print('remapping port', node_obj[1], remap_rule['to_port'])
-                    node_obj[1] = remap_rule['to_port']
-
+                    # Try to remap port only when host is remapped. Port cannot be looked at in isolation
+                    # from host level evaluation
+                    # The port value is always an integer
+                    if 'from_port' in remap_rule and 'to_port' in remap_rule:
+                        if remap_rule['from_port'] == node_obj[1]:
+                            # print('remapping port', node_obj[1], remap_rule['to_port'])
+                            node_obj[1] = remap_rule['to_port']
+                # Since these are like evaluation rules, we can stop when the first one "fires"
+                break
```

Comment:
@Grokzen I was thinking more about this and there might be one (IMO extreme) edge case where this break might cause issues. 
That is if the mapping needs happen on the same pair of hosts but different ports.

So for instance the following configuration:
```
host_port_remap=[
        {'from_host': '2.2.2.2', 'from_port': 18000, 'to_host': '127.0.0.1', 'to_port': 17000},
        {'from_host': '2.2.2.2,  'from_port': 18001, 'to_host': '127.0.0.1', 'to_port': 17001},
        {'from_host': '2.2.2.2', 'from_port': 18002, 'to_host': '127.0.0.1', 'to_port': 17002},
        {'from_host': '2.2.2.2', 'from_port': 18003, 'to_host': '127.0.0.1', 'to_port': 17003},
        {'from_host': '2.2.2.2', 'from_port': 18004, 'to_host': '127.0.0.1', 'to_port': 17004},
        {'from_host': '2.2.2.2', 'from_port': 18005, 'to_host': '127.0.0.1', 'to_port': 17005}
    ]
``` 

- One solution is to move the `break` under the innermost `if` block. So break only if **both** the host and port get re-mapped ,at which point, there is no ambiguity (I like this idea). 
- Or we can remove the `break` all together and and go through all the rules. (In which case, the last matching rule will apply)

Let me know what you think...

Reply:
Or this can be subsequently redesigned to require both host and port under the (from/to) parts of the rules (thus, giving us a unique pointer to each redis process) and go with something like
```
if remap_rule['from_host'] in node_obj[0] and remap_rule['from_port'] == node_obj[1]:
    <only then do re-mapping>
```

Reply:
I think that if i need to choose between stopping at the first rule or go through all of them and pick the last rule, i would opt not to break at the first hit but to run through all the rules and let the last one win and let the user order all the remapping rules where the last rule is applied. One interesting case you could technically do with that is that you rebind A -> B, but then in the next rule right after you rebind B -> C. I dont know who could really want to do that, but that gives more flexibility to do wonky stuff if one wants :)

Reply:
For now I agree we should remove the break. That should probably keep the edge cases to the minimum. I will push that change here today or tomorrow. We can maybe tackle more complex changes in the next iteration.